examples/asset/remote_asset_loader.rs

```diff
@@ -0,0 +1,167 @@
+    reader: Box<dyn AssetReader>,
+    writer: Box<dyn AssetWriter>,
+}
+
+impl RemoteAssetLoader {
+    fn new() -> Self {
+        // Create the root directory if it doesn't exists on local disk.
+        let create_root_dir = true;
+        let writer = AssetSource::get_default_writer("assets".to_string())(create_root_dir)
+            .expect("Current platform doesn't support asset writing.");
+        let reader = AssetSource::get_default_reader("assets".to_string())();
+
+        // This is the CDN which will be downloaded the font when missing localy.
+        let cdn_prefix =
+            "https://github.com/google/fonts/raw/main/ofl/protestrevolution/".to_string();
+
+        RemoteAssetLoader {
+            reader,
+            writer,
+            cdn_prefix,
+        }
+    }
+
+    /// Remote download the asset from CDN.
+    async fn read_remote<'a>(
+        &'a self,
+        path: &'a Path,
+    ) -> Result<Box<dyn AsyncRead + Send + Sync + Unpin + 'a>, AssetReaderError> {
+        // Only the asset name is needed, not the folder.
+        let asset_name = path.file_name().unwrap().to_str().unwrap();
+        let mut url = self.cdn_prefix.clone();
+        url.push_str(asset_name);
+
+        // A simple GET request is used, but you could set custom headers, auth and so on.
+        let request = ehttp::Request::get(url);
+
+        let body = match ehttp::fetch_async(request).await {
+            Ok(response) => {
+                // Since this is an example, only check for 200 status, but in a real world use
+                // it would be wise to check for others 2xx or 3xx status.
+                if response.status != 200 {
+                    return Err(AssetReaderError::HttpError(response.status));
+                }
+
+                response.bytes
+            }
+            Err(error) => {
+                warn!("Failed to read remote asset: {error}");
+                return Err(AssetReaderError::HttpError(500));
+            }
+        };
+
+        // Try to save the downloaded asset on disk.
+        if let Err(AssetWriterError::Io(error)) = self.writer.write_bytes(path, &body).await {
+            return Err(AssetReaderError::Io(Arc::new(error)));
+        }
+
+        // Try to read again the asset, since its saved on disk now.
+        self.reader.read(path).await
+    }
+}
+
+impl AssetReader for RemoteAssetLoader {
+    fn read<'a>(
+        &'a self,
+        path: &'a Path,
+    ) -> BoxedFuture<'a, Result<Box<Reader<'a>>, AssetReaderError>> {
+        Box::pin(async move {
+            // First check if the default reader will find the asset local. If it doesn't find
+            // then try to read from CDN.
+            match self.reader.read(path).await {
+                Ok(reader) => Ok(reader),
+                Err(error) => match error {
+                    AssetReaderError::NotFound(_) => {
+                        debug!("Asset {path:?} not found on local disk. Downloading from CDN.");
```

Comment:
This does not show up in the console for me, and for an example I think it would be nice if it did, so it is easier for a user to see what is going on. I believe you need to set the log level and/or filter: https://bevy-cheatbook.github.io/fundamentals/log.html

Reply:
Yes, usually I set my logs via `RUST_LOG` that's why I missed that, but since this is an example it make sense to setup this on app itself.